File: QuickTalk.Api/DTOs/MessageDTOs.cs
namespace QuickTalk.Api.DTOs
{
    public class SendMessageRequest
    {
        public string Content { get; set;}
        public int ChatRoomID { get; set;}
    }

    public class MessageDTO
    {
        public int MessageID { get; set; }
        public string Content { get; set; }
        public DateTime SentAt { get; set; }
        public UserDTO? Sender { get; set; }
    }
}

File: QuickTalk.Web/src/dto/MessageDTOs.js
export const createSendMessageRequestDto = (content, chatRoomID) => ({
    content: content,
    chatRoomID: chatRoomID,
});
  
export const createMessageDto = (messageID, content, sentAt, sender) => ({
    messageID: messageID,
    content: content,
    sentAt: sentAt,
    sender: sender, // UserDTO object for the sender
});